### PR TITLE
OCPBUGSM-30179: Remove unused fields in CRDs

### PR DESCRIPTION
**Pull request by rollandf**
# Description

Removing fields that are not used and won't be implemented also in
following versions.

Signed-off-by: Fred Rolland <frolland@redhat.com>

# What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [ ] None

# How was this code tested?

Please, select one or more if needed:

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

It's recommended to take a few extra minutes to provide more information about
how this code was tested. Here are some questions that may be worth answering:

- Should this PR be tested by the reviewer?
- Is this PR relying on CI for an e2e test run?
- Should this PR be tested in a specific environment?
- Any logs, screenshots, etc that can help with the review process?

# Assignees

Please, add one or two reviewers that could help review this PR.

/assign @avishayt 
/assign @mhrivnak 
/assign @filanov 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/controller/api/v1beta1/agent_types.go [hunk 1]
```diff
@@ -168,54 +130,21 @@ type HostProgressInfo struct {
...
-	Bootstrap             bool                             `json:"bootstrap,omitempty"`
-	DiscoveryAgentVersion string                           `json:"discoveryAgentVersion,omitempty"`
-	Inventory             HostInventory                    `json:"inventory,omitempty"`
-	ValidationInfo        HostValidationInfo               `json:"hostValidationInfo,omitempty"`
-	Progress              HostProgressInfo                 `json:"progress,omitempty"`
-	Connectivity          []HostConnectivityValidationInfo `json:"connectivity,omitempty"`
-	APIVipConnectivity    bool                             `json:"apiVIPConnectivity,omitempty"`
-	NtpSources            []HostNTPSources                 `json:"ntpSources,omitempty"`
-	Conditions            []conditionsv1.Condition         `json:"conditions,omitempty"`
+	State      string                   `json:"state,omitempty"`
```
Comment (filanov) [inline internal/controller/api/v1beta1/agent_types.go:140]: doesn't it move to debug info?

Comment (rollandf) [reply internal/controller/api/v1beta1/agent_types.go:140]: Yes, I will rebase on top of #1972 once merged

Comment (rollandf) [reply internal/controller/api/v1beta1/agent_types.go]: Done

**Review by filanov — APPROVED** [review]

**Comment by tsorya** [comment]
nice one :) , always nice to see code removed

**Comment by rollandf** [comment]
Cherry pick: #1987

**Pull request merged**